### PR TITLE
Batch l2 gradients: extensions 

### PR DESCRIPTION
**Pull request by sbharadwajj**
1.  BatchL2Grad extensions for `Conv1d`, `Conv3d`, `ConvTranspose1d`, `ConvTranspose2d`, `ConvTranspose3d`
2. Test suite for the same

### CODE REVIEW

File: backpack/utils/conv_transpose.py [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
 
+from backpack.utils.ein import eingroup
+
+
+def get_convtranspose1d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
```
Comment (f-dangel) [inline backpack/utils/conv_transpose.py:12]: The `torch.flatten(...)` should not be required. Please check and remove it if possible

File: backpack/utils/conv_transpose.py [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
+    dE_dY = eingroup("n,c,h,w->n,c,hw", grad_out)
+    return X, dE_dY
+
+
+def get_convtranspose3d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
```
Comment (f-dangel) [inline backpack/utils/conv_transpose.py:36]: The `torch.flatten(...)` should not be required. Please check and remove it if possible

File: backpack/utils/conv_transpose.py [hunk 3]
```diff
@@ -1,6 +1,44 @@
...
+    dE_dY = eingroup("n,c,l->n,c,l", grad_out)
+    return X, dE_dY
+
+
+def get_convtranspose2d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
```
Comment (f-dangel) [inline backpack/utils/conv_transpose.py:24]: The `torch.flatten(...)` should not be required. Please check and remove it if possible

File: backpack/utils/conv_transpose.py [hunk 4]
```diff
@@ -1,6 +1,44 @@
...
+
+def get_convtranspose1d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
+        N, C_in * kernel_size_numel, -1
+    )
+    dE_dY = eingroup("n,c,l->n,c,l", grad_out)
```
Comment (f-dangel) [inline backpack/utils/conv_transpose.py:15]: That call seems to do nothing. Please check and remove if possible

File: test/extensions/implementation/autograd.py [hunk 5]
```diff
@@ -29,3 +29,10 @@ def batch_grad(self):
                 batch_grads[idx][b, :] = g.detach() * factor
 
         return batch_grads
+
+    def batch_l2_grad(self):
+        batch_grad = self.batch_grad()
+        batch_l2_grads = [
+            (g ** 2).sum(list(range(1, len(g.shape)))) for g in batch_grad
```
Comment (f-dangel) [inline test/extensions/implementation/autograd.py:36]: The summation over all axes except the `0`th could also be written as `(g ** 2).flatten(start_dim=1).sum(1) for g in ...`. Do you think that this way the code would be easier to read?

File: backpack/utils/conv_transpose.py (hunk 4 above)
Comment (sbharadwajj) [reply backpack/utils/conv_transpose.py:15]: yes. But I suggest we still return `grad_out` as `dE_dY` to maintain uniformity across other functions. What do you suggest?

File: test/extensions/implementation/autograd.py (hunk 5 above)
Comment (sbharadwajj) [reply test/extensions/implementation/autograd.py]: Yes, this is a better way to write it
